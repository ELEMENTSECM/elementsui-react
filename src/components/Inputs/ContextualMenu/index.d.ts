import * as React from 'react';

export interface ContextualMenuMenuProps {
        items?: any[];
    }

    export interface ContextualMenuProps {
        /**
         * ContextualMenu primary
         */
        isPrimary?: boolean;
        /**
         * ContextualMenu button text
         */
        buttonText?: string;
        /**
         * ContextualMenu button ariaLabel
         */
        ariaLabel?: string;
        /**
         * ContextualMenu menuitems
         */
        menuProps: ContextualMenuMenuProps;
    }

    export const ContextualMenu: React.SFC<ContextualMenuProps>;


