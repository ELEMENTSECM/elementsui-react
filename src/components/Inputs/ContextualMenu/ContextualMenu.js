import * as React from 'react';
import PropTypes from 'prop-types';
import { DefaultButton } from 'office-ui-fabric-react/lib/Button';

/** Contextual Menu Button */
function ContextualMenu({ isPrimary, ariaLabel, buttonText, menuProps }) {
	return (
		<DefaultButton primary={isPrimary} ariaLabel={ariaLabel} text={buttonText} menuProps={menuProps} />
	);
}

ContextualMenu.propTypes = {
	/** ContextualMenu primary  */
	isPrimary: PropTypes.bool,
	/** ContextualMenu button text */
	buttonText: PropTypes.string,
	/** ContextualMenu button ariaLabel */
	ariaLabel: PropTypes.string,
	/**  ContextualMenu menuitems */
	menuProps: PropTypes.shape({
		items: PropTypes.array
	}).isRequired
};

export default ContextualMenu;
