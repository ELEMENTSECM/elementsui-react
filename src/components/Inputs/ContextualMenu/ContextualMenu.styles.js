export const styles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-ContextualMenu',
			{
				// place your styles here
			},
			className
		]
	};
};
