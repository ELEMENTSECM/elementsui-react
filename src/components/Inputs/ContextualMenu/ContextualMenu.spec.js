import React from 'react';
import renderer from 'react-test-renderer';
import ContextualMenu from './ContextualMenu';

describe('ContextualMenu', () => {
	test('displays `Foo` as text', () => {
		const tree = renderer.create(<ContextualMenu htmlId="contextualMenuButton" text="Foo" />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe('ContextualMenuProps', () => {
	const items = [
		{
			key: 'newItem',
			name: 'New item',
			onClick: () => console.log('New item clicked')
		},
		{
			key: 'noTarget',
			name: 'Link default',
			href: 'http://evry.com'
		}
	];

	test('displays menu correctly', () => {
		const tree = renderer
			.create(<ContextualMenu id="testContextualMenu" text="Contextual Menu" menuProps={{ items }} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
