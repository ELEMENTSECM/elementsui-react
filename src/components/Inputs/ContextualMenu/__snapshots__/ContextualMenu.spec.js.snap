// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextualMenu displays \`Foo\` as text 1`] = `
<button
  aria-describedby={null}
  aria-label={undefined}
  aria-labelledby={null}
  aria-pressed={undefined}
  className="ms-Button ms-Button--default root-36"
  data-is-focusable={true}
  disabled={undefined}
  type="button"
>
  <div
    className="ms-Button-flexContainer flexContainer-37"
  />
</button>
`;

exports[`ContextualMenuProps displays menu correctly 1`] = `
<button
  aria-describedby={null}
  aria-expanded={false}
  aria-haspopup={true}
  aria-label={undefined}
  aria-labelledby={null}
  aria-owns={null}
  aria-pressed={undefined}
  className="ms-Button ms-Button--default root-36"
  data-is-focusable={true}
  disabled={undefined}
  onClick={[Function]}
  onKeyDown={[Function]}
  type="button"
>
  <div
    className="ms-Button-flexContainer flexContainer-37"
  >
    <i
      aria-hidden={true}
      className="root-44 ms-Button-menuIcon menuIcon-41"
      data-icon-name="ChevronDown"
      role="presentation"
    />
  </div>
</button>
`;
